ui/packages/components/src/DatePicker/DateTimeInput.tsx

```diff
@@ -0,0 +1,329 @@
+import { useEffect, useRef, useState, type MutableRefObject } from 'react';
+import { format as formatDate, getDaysInMonth, isValid, parseISO } from 'date-fns';
+import { ZodError, type ZodSchema } from 'zod';
+
+import { cn } from '../utils/classNames';
+import {
+  getHourSchema,
+  millisecondsSchema,
+  minutesSchema,
+  monthSchema,
+  periodSchema,
+  secondsSchema,
+  yearSchema,
+} from './timeSchemas';
+
+type HandleTimeChangeProps = {
+  e: React.ChangeEvent<HTMLInputElement>;
+  schema: ZodSchema<any>;
+  setInputValue: React.Dispatch<React.SetStateAction<string>>;
+};
+
+type TimeInputProps = {
+  is24HourFormat?: boolean;
+  selectedDateTime?: Date;
+  onSelect: React.Dispatch<React.SetStateAction<Date | undefined>>;
+  setValid: React.Dispatch<React.SetStateAction<boolean>>;
+  valid: boolean;
+  error?: string;
+};
+
+const isNumeric = (num: any) =>
+  (typeof num === 'number' || (typeof num === 'string' && num.trim() !== '')) &&
+  !isNaN(num as number);
+
+const formatWithDefault = (format: string, defaultValue: string, date?: Date) =>
+  date ? formatDate(date, format) : defaultValue;
+
+export function DateTimeInput({
+  selectedDateTime,
+  onSelect,
+  is24HourFormat = false,
+  setValid,
+  valid,
+}: TimeInputProps) {
+  const daysRef = useRef<HTMLInputElement | null>(null);
```

Comment:
```suggestion
  const daysRef = useRef<HTMLInputElement>(null);
```
You don't need to specify the null